lib/internal/async_hooks.js

```diff
@@ -371,6 +374,8 @@ function emitAfterScript(asyncId) {
   if (async_hook_fields[kAfter] > 0)
     emitAfterNative(asyncId);
 
+  setExecutionAsyncResource(null);
```

Comment:
see https://github.com/nodejs/node/pull/21313/files?file-filters%5B%5D=.js#r195097421
Isn't a stack needed here?

Reply:
Yes, I think so. I’m going to be working on this more starting later today, so I’ll try and get these issues sorted out soon. 🙂

Reply:
Done, just trying to figure out a failure in the `parallel/test-heapdump-async-hooks-init-promise.js` test now.